### ISSUE TITLE
Prefer `npci=0x2000` boot arg vs `Above 4G decoding` in BIOS

### ISSUE THREAD

**Issue by laurynas-karvelis** [dortania/OpenCore-Desktop-Guide#38]
Hello,

First of all I wanted to thank everyone involved working on this project! It's an unbelievable achievement!

Currently I have a dual boot setup (Windows and MacOS) on a AMD platform. Respecting OpenCore recommendations to turn on `Above 4G decoding` BIOS setting I've been able to successfully boot into MacOS and use it without any problems. As my motherboard is MSI Tomahawk Max B450 I've noticed that keeping `Above 4G decoding` on and booting into Windows will cause built in ethernet controller not working at all. So, before each PC boot I had to toggle `Above 4G decoding` BIOS option.

Recently I've noticed the [AMD BIOS Settings](https://dortania.github.io/OpenCore-Desktop-Guide/AMD/zen.html#amd-bios-settings) section mentioning `npci=0x2000` as an alternative to those who cannot find `Above 4G decoding` setting in their BIOS.

To my surprise, disabling `Above 4G decoding` and adding `npci=0x2000` boot arg inside the config.plist not only removed a necessity to constantly toggle `Above 4G decoding` BIOS option, but made MacOS actually feel more snappy and responsive!

Summary of this all: would it not be a better idea to recommend AMD Zen users to leave `Above 4G decoding` DISABLED, but add `npci=0x2000` boot arg instead? It seems it might fix potential issues for dual boot users and *potentially* improve GPU performance on MacOS too.

My PC configuration as follows:
* Motherboard: MSI Tomahawk Max B450
* CPU: AMD Ryzen 7 3700X
* GPU: Radeon RX 5700 XT
* RAM: 32 GB Kingston DDR4 3333MHz
* OS: macOS Catalina 10.15.4 19E287 x86_64

Dual booting using rEFInd Boot Manager.

**Comment by khronokernel** [comment]
> Summary of this all: would it not be a better idea to recommend AMD Zen users to leave Above 4G decoding DISABLED, but add npci=0x2000 boot arg instead? It seems it might fix potential issues for dual boot users and potentially improve GPU performance on MacOS too.

Well the big problem is npci is very hit or miss on whether it works or not, as it's a very "hacky" fix since we're just trying to get around AMD's poor implementation with PCI IRQ by not enabling certain debugging which avoids a rebuild step that breaks everything. But certain firmwares still have issues even with this flag but is easily fixed with Above4G(on newer firmwares, in general early AMD motherboard firmwares are a total shit show for macOS. And some OEMs are still borked)

So regarding Windows, more AMD issues but also a bit of a fault on the drivers(but I have a good hunch it's the mobo at fault again not reporting it correctly) as devs will just assume devices will always be within the 4GB memory space and not account for cases where it may be above(Since what "normal" consumer would bother with these kinds of settings in the BIOS). So in this case when your NIC is forced into 4GB+ territory, the drivers can't find it/communicate properly.

* The hunch that it's a firmware issue and less of a driver issue is that the RTL8111H chipset has been a known good controller on Intel, I've had it myself on an old Intel board and Above4G made no difference

So TL;DR: 
* npci is hacky and should be avoided unless explicitly needed
  * Tbf AMD on macOS in general is hacky so guess it fits well
* Above4G breaking windows is a firmware bug
  * Maybe writing a nice letter to MSI might help ;p 
  * AsRock is known to have this same bug so likely something in the AGESA
* Find what works for you, preferably Above4G but if npci works so be it

> Dual booting using rEFInd Boot Manager.

Just a heads up, avoid rEFInd when using OpenCore as it's an unsupported configuration and known to actually break macOS booting in many cases

**Comment by dhinakg** [comment]
Fwiw, this is the response I got from ASRock:
![image](https://user-images.githubusercontent.com/17605561/81324041-642cac00-9064-11ea-9838-6fb19786fc62.png)
